packages/neo-one-smart-contract-lib/src/ownership/Ownable.ts

```diff
@@ -0,0 +1,52 @@
+import { Address, constant, createEventNotifier, SmartContract } from '@neo-one/smart-contract';
+
+/**
+ * @title Ownable
+ * @dev A ownable contract has an owner address
+ */
+
+/* tslint:disable-next-line: variable-name */
+const transferred_ownership = createEventNotifier<Address | undefined, Address>('transferred ownership', 'from', 'to');
+
+export function Ownable<TBase extends Constructor<SmartContract>>(base: TBase) {
+  class OwnableClass extends base {
+    protected mutableOwner: Address | undefined;
+    public constructor(...args: any[]) {
```

Comment:
Mixins shouldn't use constructor arguments because there's no guarantee of the order or what the arguments are going to be. In fact, you shouldn't have a constructor at all.

Reply:
Abstract it becomes.  👍